### PR TITLE
[enhancement] arrayColumn implements assign()

### PR DESCRIPTION
**Pull request by fzhedu**
Signed-off-by: fzhedu <fzhedu@gmail.com>

## What type of PR is this：
- [ ] bug
- [ ] feature
- [x] enhancement
- [ ] refactor
- [ ] others

## Which issues of this PR fixes ：

Fixes #

arrayColumn implements assign(), which is used when unfold const array columns.

## Checklist:

- [x] I have added test cases for my bug fix or my new feature
- [ ] I have added user document for my new feature or new function

### CODE REVIEW

File: be/src/column/array_column.cpp [hunk 1]
```diff
@@ -66,7 +66,10 @@ void ArrayColumn::resize(size_t n) {
 }
 
 void ArrayColumn::assign(size_t n, size_t idx) {
-    DCHECK(false) << "array column shouldn't call assign";
+    DCHECK_LE(idx, this->size()) << "Range error when assign arrayColumn.";
+    auto datum = get(idx);
+    this->reset_column();
```
Comment (imay) [inline be/src/column/array_column.cpp:71]: When this column is reset is the datum still valid? IMO datum is a reference not a deep copy obj

Comment (fzhedu) [reply be/src/column/array_column.cpp]: fix that by copy the datum to a new column then swap.

**Review by kangkaisen — APPROVED** [review]

**Comment by fzhedu** [comment]
run starrocks_admit_test

**Review by stdpain — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Pull request merged**